addons/binding/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/handler/RokuHandler.java

```diff
@@ -0,0 +1,125 @@
+            try {
+                rokuDevice.generateAction(channelUID);
+            } catch (IOException e) {
+                logger.error("IOException occurred when attempting action", e);
+            }
+            updateState(channelUID, OnOffType.OFF);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing RokuDevice handler '{}'", getThing().getUID());
+        Number refreshInterval = (Number) this.getConfig().get(REFRESH_INTERVAL);
+        scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                updateData();
+            }
+        }, 0, refreshInterval.longValue(), TimeUnit.SECONDS);
+        logger.debug("Polling job scheduled to run every {} sec. for '{}'", refreshInterval, getThing().getUID());
+    }
+
+    private synchronized void updateData() {
+        logger.info("Update Roku data '{}'", getThing().getUID());
+        String ipAddress = (String) this.getConfig().get(IP_ADDRESS);
+        Number port = (Number) this.getConfig().get(PORT);
+        RokuState state = new RokuState(new RokuCommunication(ipAddress, port));
+        try {
+            state.updateDeviceInformation();
+        } catch (IOException e) {
+            logger.debug("Roku device '{}' but is not communicating", ipAddress, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.toString());
+        }
+
+        updatePropertiesFromState(state);
+
+        updateState(CHANNEL_ACTIVE, state.getActive());
+        updateState(CHANNEL_HEADPHONES, new StringType(state.getHeadphonesConnected()));
+        updateState(CHANNEL_ICON, state.getActiveImage());
+        /**
+         * Future Functionality
+         * updateState(CHANNEL_APPBROWSER, state.getApplicationMenu());
+         */
+        if (!state.getPowerMode().equals("PowerOn")) {
+            updateStatus(ThingStatus.OFFLINE);
+        } else {
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    private void updatePropertiesFromState(RokuState state) {
+        Map<String, String> properties = editProperties();
+        properties.put(PROPERTY_UDN, state.getUdn());
+        properties.put(Thing.PROPERTY_SERIAL_NUMBER, state.getSerialNumber());
+        properties.put(PROPERTY_DEVICE_ID, state.getDeviceId());
+        properties.put(Thing.PROPERTY_VENDOR, state.getVendorName());
+        properties.put(PROPERTY_MODEL_NAME, state.getModelName());
+        properties.put(Thing.PROPERTY_MODEL_ID, state.getModelNumber());
+        properties.put(PROPERTY_MODEL_REGION, state.getModelRegion());
+        if (state.getNetworkType().equals("ethernet")) {
+            properties.put(PROPERTY_MAC, state.getEthernetMac());
+        } else {
+            properties.put(PROPERTY_MAC, state.getWifiMac());
+        }
+        if (state.getUserDeviceName() != null && !state.getUserDeviceName().equals("")) {
```

Comment:
If you want this check consider using `StringUtils.isNotEmpty()` just search for it in our code, we have apache commons 2.6 available by default.